lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java

```diff
@@ -757,6 +808,77 @@ Duration calculateDuration(
     return result;
   }
 
+  private void addErrorsToResultObject(
+      String owner,
+      BulkOperationResults<String, TaskanaException> bulkLog,
+      List<MinimalTaskSummary> existingMinimalTaskSummaries) {
+    for (MinimalTaskSummary taskSummary : existingMinimalTaskSummaries) {
+      if (!owner.equals(taskSummary.getOwner())) { // owner was not set
+        if (!taskSummary.getTaskState().equals(TaskState.READY)) { // due to invalid state
+          bulkLog.addError(
+              taskSummary.getTaskId(),
+              new InvalidStateException(
+                  "Task "
+                      + taskSummary.getTaskId()
+                      + " is in state "
+                      + taskSummary.getTaskState()));
+        } else { // due to unknown reason
+          bulkLog.addError(
+              taskSummary.getTaskId(),
+              new UpdateFailedException(
+                  "Could not set owner of Task " + taskSummary.getTaskId() + "."));
+        }
+      }
+    }
+  }
+
+  private void handleNonExistingTasks(
+      List<String> taskIds,
+      List<MinimalTaskSummary> existingMinimalTaskSummaries,
+      BulkOperationResults<String, TaskanaException> bulkLog) {
+    List<String> nonExistingTaskIds = new ArrayList<>(taskIds);
+    List<String> existingTaskIds =
+        existingMinimalTaskSummaries.stream()
+            .map(MinimalTaskSummary::getTaskId)
+            .collect(Collectors.toList());
+    nonExistingTaskIds.removeAll(existingTaskIds);
+    for (String taskId : nonExistingTaskIds) {
+      bulkLog.addError(taskId, new TaskNotFoundException(taskId, "Task was not found"));
+    }
+  }
+
+  private List<String> filterForAuthorized(
+      List<String> taskIds, BulkOperationResults<String, TaskanaException> bulkLog) {
+    List<String> accessIds = getAccessIds();
+    List<String> tasksAuthorizedFor = new ArrayList<>(taskIds);
+    if (accessIds != null) {
+      List<String> tasksNotAuthorizedFor =
+          taskMapper.filterTaskIdsNotAuthorizedFor(taskIds, accessIds);
+      tasksAuthorizedFor.removeAll(tasksNotAuthorizedFor);
+      String currentUserId = CurrentUserContext.getUserid();
+      for (String taskId : tasksNotAuthorizedFor) {
+        bulkLog.addError(
+            taskId,
+            new NotAuthorizedException(
+                "Current user not authorized for task " + taskId, currentUserId));
+      }
+    }
+    return tasksAuthorizedFor;
+  }
+
+  private List<String> getAccessIds() {
```

Comment:
This would be better placed in TaskanaEngine or even better in CurrentUserContext?

Reply:
Ok, I checked - lowercasing is done in CurrentUserContext. I removed it from TaskServiceImpl